qp/factory.py

```diff
@@ -194,6 +196,50 @@ def pretty_print(self, stream=sys.stdout):
             stream.write("%s: %s\n" % (class_name, cl))
             cl.print_method_maps(stream)
 
+    @staticmethod
+    def concatenate(ensembles):
+        """Concatanate a list of ensembles
+
+        Parameters
+        ----------
+        ensembles : `list`
+            The ensembles we are concatanating
+
+        Returns
+        -------
+        ens : `qp.Ensemble`
+            The output
+        """
+        if not ensembles:  #pragma: no cover
+            return None
+        metadata_list = []
+        objdata_list = []
+        ancil_list = []
+        gen_func = None
+        for ensemble in ensembles:
+            metadata_list.append(ensemble.metadata())
+            objdata_list.append(ensemble.objdata())
+            if gen_func is None:
+                gen_func = ensemble.gen_func
+            if ancil_list is not None:
+                if ensemble.ancil is None:
+                    ancil_list = None
+                else:  #pragma: no cover
+                    ancil_list.append(ensemble.ancil)
+        if not compare_dicts(metadata_list):  #pragma: no cover
```

Comment:
Same exact issue as pops up in `append` in ensemble.py l260 with comparing dictionaries where at least one of the elements is an array.